projects/01-rubygems-cli/solution/lib/ruby_gems_api.rb

```diff
@@ -0,0 +1,33 @@
+require "faraday"
+require "faraday/net_http"
+require "json"
+require "./errors/gem_not_found"
+require "./errors/bad_request"
+
+class RubyGemsApi
+  @connection = Faraday.new("https://rubygems.org") do |faraday|
+    faraday.adapter :net_http
+  end
+
+  class << self
+    def gem_info(gem_name)
+      raise ArgumentError.new("Invalid gem name provided.") if gem_name.nil? || gem_name.empty?
+      
+      response = @connection.get("/api/v1/gems/#{gem_name}.json")
+      raise GemNotFound.new("Gem #{gem_name} not found.") if response.body == "This rubygem could not be found."
```

Comment:
Messages should not be used programmatically because they are easy to change and then your logic breaks. 
Ideally you want to base your logic on something that is less likely to change (as in the HTTP status code here, which is 404 -- Not Found).

Reply:
4cfd6b59d3900c3186840aa7ff6d8d57ada55251